ethers-solc/src/compile/output/mod.rs

```diff
@@ -449,7 +443,7 @@ impl AggregatedCompilerOutput {
         })
     }
 
-    pub fn diagnostics<'a>(&'a self, ignored_error_codes: &'a [u64], warnings_as_errors: bool) -> OutputDiagnostics {
+    pub fn diagnostics<'a>(&'a self, ignored_error_codes: &'a [u64], warnings_as_errors: &'a Severity) -> OutputDiagnostics {
```

Comment:
no need to take this as ref
```suggestion
    pub fn diagnostics<'a>(&'a self, ignored_error_codes: &'a [u64], warnings_as_errors: Severity) -> OutputDiagnostics {
```